messenger/http_transporter_test.go

```diff
@@ -266,6 +267,34 @@ func TestTransporterStartAndStop(t *testing.T) {
 	}
 }
 
+func TestMutadedHostUPid(t *testing.T) {
+	serverId := "testserver"
+	serverPort := getNewPort()
+	serverHost := "127.0.0.1"
+	serverAddr := serverHost + ":" + strconv.Itoa(serverPort)
+
+	// override the upid.Host with this listener IP
+	addr := net.ParseIP("127.0.1.1")
+
+	// setup receiver (server) process
+	uPid, err := upid.Parse(fmt.Sprintf("%s@%s", serverId, serverAddr))
+	assert.NoError(t, err)
+	receiver := NewHTTPTransporter(uPid, addr)
+
+	errch := receiver.Start()
```

Comment:
- why call Start/Stop instead of listen(), which is the func that you really changed?
- this is great for testing the code as it was *before* your changes. how about adding a test case that stresses the code you added to HTTPTransporter?